adapters/triplelift/triplelift.go

```diff
@@ -0,0 +1,147 @@
+package triplelift
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type TripleliftAdapter struct {
+	endpoint string
+}
+
+type TripleliftInnerExt struct {
+	Format int `json:"format"`
+}
+
+type TripleliftRespExt struct {
+	Triplelift TripleliftInnerExt `json:"triplelift_pb"`
+}
+
+func getBidType(ext TripleliftRespExt) openrtb_ext.BidType {
+	t := ext.Triplelift.Format
+	if t == 11 {
+		return openrtb_ext.BidTypeVideo
+	}
+	return openrtb_ext.BidTypeBanner
+}
+
+func processImp(imp *openrtb.Imp) error {
+	// get the triplelift extension
+	var ext adapters.ExtImpBidder
+	var tlext openrtb_ext.ExtImpTriplelift
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(ext.Bidder, &tlext); err != nil {
+		return err
+	}
+	if imp.Banner == nil && imp.Video == nil {
+		return fmt.Errorf("neither Banner nor Video object specified")
+	}
+	imp.TagID = tlext.InvCode
+	// floor is optional
+	if tlext.Floor == nil {
```

Comment:
Could just have `if tlext.Floor != nil { imp.BidFloor = *tlext.Floor }` as you return `nil` in either case

Reply:
ah good point, that's a bit more parsimonious.  but it seems this is already merged. if I have to patch later I'll take this idea and apply it though.